lib/pdf-editor-view.js

```diff
@@ -490,6 +495,32 @@ export default class PdfEditorView extends ScrollView {
     }, () => this.finishUpdate());
   }
 
+  computeMaxPageWidth(){
+    Promise.all(
+      _.range(1, this.pdfDocument.numPages + 1).map((pdfPageNumber) =>
+        this.pdfDocument.getPage(pdfPageNumber).then((pdfPage) =>
+          pdfPage.getViewport(1.0).width
+        )
+      )
+    ).then((pdfPageWidths) => {
+      this.maxPageWidth = Math.max(...pdfPageWidths);
+      this.zoomFit();
```

Comment:
It feels kind of strange that we are calling a zoomFit from within a function called `computeMaxPageWidth` -- it's not obvious that this function would do anything more than compute the max page width.

How do you feel about not calling zoomFit here, and instead within the zoomFit function you would call computeMaxPageWidth, wait for that to complete, and then continue the work in zoomFit? Currently, in zoomFit, you have

```
+    if (this.maxPageWidth == 0) {
+      this.computeMaxPageWidth();
+      return;
+    }
```

which means you call computeMaxPageWidth from within zoomFit, and then computeMaxPageWidth calls zoomFit. And the jumping around seems unnecessary and a bit confusing.

Reply:
zoomFit is called again to make zoomfit effective in case maxPageWidth is 0 (ie either not computed yet or resetted). I change the methodname to computeMaxPageWidthAndTryZoomFit.